hooks/push.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+use PHPMailer\PHPMailer\PHPMailer;
+use PHPMailer\PHPMailer\Exception;
+
+/**
+ * GitHub webhook to send emails using push event
+ */
+
+error_reporting(E_ALL);
+
+define('PMAHOOKS', True);
+
+require_once(__DIR__.'/lib/github.php');
+
+if ( !is_file(__DIR__.'/vendor/autoload.php')) {
+    fail("Run composer install");
+    exit;
+}
+
+//Load Composer's autoloader
+require_once(__DIR__.'/vendor/autoload.php');
+
+github_verify_post();
+
+
+$inputData = json_decode(file_get_contents('php://input'));
+
+if (json_last_error() !== 0) {
+    fail("JSON body was not well formed.");
+    exit;
+}
+
+
+$data = gihub_webhook_push($inputData);
+
+
+$mail = new PHPMailer(true);
+try {
+
+    $mail->SMTPDebug  = 0;
+    $mail->isSMTP();
+    $mail->Host       = SMTP_HOST;
+    $mail->SMTPAuth   = true;
+    $mail->Username   = SMTP_USERNAME;
+    $mail->Password   = SMTP_PASSWORD;
+    $mail->SMTPSecure = SMTP_MODE;
+    $mail->Port       = SMTP_PORT;
+
+    $mail->setFrom(SMTP_SEND_FROM, $data->authorName);
```

Comment:
Replacing `SMTP_SEND_FROM` with `$data->authorEmail` to match what exists, Is it a good idea ?

Reply:
AFAIK this is how current hook works.

Reply:
Yes, maybe, I find it strange that emails are send by your email and sent by mailing list but do not trigger SPF.
Maybe I did not understand everything with how works a mailing list.

Name | Value
-- | --
Message-ID | <5b55b6e498f08_b6c2ac949a46f7c42864 [EDITED] hookshot-fe-6b2eebc.cp1-iad.github.net.mail>
Date | Mon, 23 Jul 2018 04:07:16 -0700
From | Michal =?UTF-8?B?xIxpaGHFmQ==?= <michal [EDITED] cihar.com>
To | git [EDITED] phpmyadmin.net
Subject | [phpMyAdmin Git] [phpmyadmin/phpmyadmin] 7efcc1: Translated using  Weblate (Croatian)
Return-Path | <git-bounces [EDITED] phpmyadmin.net>